src/components/views/elements/SettingsFlag.tsx

```diff
@@ -108,7 +108,7 @@ export default class SettingsFlag extends React.Component<IProps, IState> {
         const label = this.props.label ?? SettingsStore.getDisplayName(this.props.name, this.props.level);
         const description = SettingsStore.getDescription(this.props.name);
         const shouldWarn = SettingsStore.shouldHaveWarning(this.props.name);
-        const disabled = this.state.disabled || !canChange;
+        const disabled = this.isSettingDisabled() || !canChange;
```

Comment:
Hmm, I'm not convinced by this fix. Isn't the point of the state that it represents the state of the component's UI? If the state is wrong, I think we'd just end up with more problems by working around it.

Reply:
I agree your point in general but in this (i guess corner) case specifically

the "disabled" state of checkbox is  established/updated on only two ocassions

1. during mounting of component i.e in constructor
2. when this particular setting gets changed... but here is the catch "useCompactLayout" settings uses an   Incompatible controller, `controller: new IncompatibleController("layout", false, (v: Layout) => v !== Layout.Group)`, and disabled value depends on "layout" settings

so anytime the "layout" settings changes the value of disabled changes but state of the checkbox doesnt know about it...thats why i put the to get correct value on every render i replaced the state with the function call.

I think the "canChange" in line 104 SettingsFlag.tsx `const canChange = SettingsStore.canSetValue(this.props.name, this.props.roomId ?? null, this.props.level);` is somewhat analogous to "disabled" where its getting value directly from SettingsStore not as state.

So in a nutshell, may be we need to handle it special or any other approach from you is also welcome. 
Thanks for the response.